apps/server/src/modules/files/service/files.service.spec.ts

```diff
@@ -125,31 +148,36 @@ describe(FilesService.name, () => {
 					}),
 					fileEntityFactory.buildWithId({
 						permissions: [yetAnotherUserPermission, userPermission, anotherUserPermission],
+						creatorId: userId,
 					}),
 					fileEntityFactory.buildWithId({
 						permissions: [anotherUserPermission, yetAnotherUserPermission, userPermission],
 					}),
 					fileEntityFactory.buildWithId({
 						permissions: [userPermission, yetAnotherUserPermission, anotherUserPermission],
+						creatorId: userId,
 					}),
 					fileEntityFactory.buildWithId({
 						permissions: [yetAnotherUserPermission, anotherUserPermission, userPermission],
 					}),
 				];
 
-				repo.findByPermissionRefId.mockResolvedValueOnce(entities);
+				repo.findByPermissionRefIdOrCreatorId.mockResolvedValueOnce(entities);
 
-				const result = await service.removeUserPermissionsToAnyFiles(userId);
+				const result = await service.removeUserPermissionsOrCreatorReferenceToAnyFiles(userId);
 
 				expect(result).toEqual(5);
 
-				entities.forEach((entity) => {
-					expect(entity.permissions).not.toContain(userPermission);
-					expect(entity.permissions).toContain(anotherUserPermission);
-					expect(entity.permissions).toContain(yetAnotherUserPermission);
-				});
+				for (let i = 0; i < entities.length; i += 1) {
+					expect(entities[i].permissions).not.toContain(userPermission);
+					if (i === 1 || i === 3) {
+						expect(entities[i]._creatorId).toBe(undefined);
+					}
+					expect(entities[i].permissions).toContain(anotherUserPermission);
+					expect(entities[i].permissions).toContain(yetAnotherUserPermission);
+				}
```

Comment:
Why should a test have so much logic?

Reply:
this test was already existing I just added to it needed data to cover all posibilities

Reply:
I already understood. But still I was expecting more separate tests than for and if statements in the tests. For me, the for and if statements are things to test again.